### PR TITLE
Rework `ExecutionError` type

### PR DESCRIPTION
**Pull request by xhjkl**
This PR changes the JSONRPC response to nest error message in field `error.message`:
```
$ curl localhost:3000/api --data-binary ...
...
* upload completely sent off: 222 out of 222 bytes
< HTTP/1.1 201 Created
...
< 
* Connection #0 to host localhost left intact
{   "jsonrpc": "2.0",
    "id": "from-curl",
    "error": { "message": "it works" }
}
```

### CODE REVIEW

File: website/api-gateway/src/json-rpc/handler.ts [hunk 1]
```diff
@@ -78,18 +78,21 @@ export class RpcMessageHandler {
...
+          } else if ('error' in value) {
+            resolve(this.getResponse(procedure, value.error));
           } else {
             resolve(this.getResponse(procedure, null, value));
           }
-        });
-      });
-    }
+        })
+        .finally(() => console.log('end'));
```
Comment (osipov-mit) [inline website/api-gateway/src/json-rpc/handler.ts:95]: I guess it's unnecessary here

File: website/data-storage/src/consumer/consumer.service.ts [hunk 2]
```diff
@@ -88,9 +88,9 @@ export class ConsumerService {
 
   async programData(params: FindProgramParams): Result<Program> {
     try {
-      return (await this.programService.findProgram(params)) || { error: new ProgramNotFound().message };
-    } catch (error) {
-      return { error: error.message };
+      return (await this.programService.findProgram(params)) || { error: { message: new ProgramNotFound().message } };
```
Comment (osipov-mit) [inline website/data-storage/src/consumer/consumer.service.ts:91]: I think it's better to move throwing this error to `findProgram` method

**Review by osipov-mit — CHANGES_REQUESTED** [review]

File: website/api-gateway/src/json-rpc/handler.ts (hunk 1 above)
Comment (xhjkl) [reply website/api-gateway/src/json-rpc/handler.ts:95]: True; I somehow thought it was here before me.

File: website/data-storage/src/consumer/consumer.service.ts (hunk 2 above)
Comment (xhjkl) [reply website/data-storage/src/consumer/consumer.service.ts:88]: Sure; thanks.

File: website/data-storage/src/programs/programs.service.ts [hunk 3]
```diff
@@ -84,7 +85,7 @@ export class ProgramsService {
       return program;
     } catch (error) {
       logger.error(error);
-      return null;
+      throw new ProgramNotFound(error.message);
```
Comment (osipov-mit) [inline website/data-storage/src/programs/programs.service.ts:88]: And one more thing
It's necessary to handle this exception in the places where this method is called. (setStatus, isInDb, addProgramInfo methods and a few more)

**Review by osipov-mit — CHANGES_REQUESTED** [review]

File: website/data-storage/src/programs/programs.service.ts (hunk 3 above)
Comment (osipov-mit) [reply website/data-storage/src/programs/programs.service.ts:88]: @xhjkl What about this?

**Pull request closed without merging**